Library/TestHelpers/MockStripeIntentService.swift

```diff
@@ -0,0 +1,51 @@
+@testable import KsApi
+@testable import Library
+import ReactiveSwift
+
+public struct StripeIntentRequestTestData: Equatable {
+  let projectId: String?
+  let pledgeTotal: Double?
+  let paymentIntentEnvelope: PaymentIntentEnvelope?
+  let context: GraphAPI.StripeIntentContextTypes?
+  let clientSeretEnvelope: ClientSecretEnvelope?
+}
+
+public class MockStripeIntentService: StripeIntentServiceType {
+  public private(set) var intentRequests = [StripeIntentRequestTestData]()
```

Comment:
Is this ever read, or only ever set? I don't see it used anywhere

Reply:
I'm appending to this array in the create methods below and then referencing it in my test assertions in `StripeIntentServiceTests`. [👀 ](https://github.com/kickstarter/ios-oss/pull/2050/files#diff-085de37a469a9dbfb6ea20e854e7a1aef83a6345c17037e4ef7841d28ee9ce6bR41)

Reply:
Ah, I see, it's used in the tests. I'm curious why you're tracking this separately instead of just using the test methods that test signals?

Reply:
The main reason is that keeping track of each signal request in an array helps us guarantee that only the expected number of requests are made and with the correct request types/data.

Do we have test helpers that do this for us? 😅

Reply:
Hm, it might just be that the tests seem a bit tautological to me - you call the mock service, then validate that the mock service was called, instead of validating its output. An alternative could be to use a mock `apiService` in your mock service, to mock out the networking calls, and then that would validate the whole networking chunk?

Reply:
How many times can I say "mock" in one comment 😆